mirrord/operator/src/crd.rs

```diff
@@ -199,3 +341,190 @@ pub struct MirrordPolicySpec {
+    pub env_updates: BTreeMap<String, QueueNameUpdate>,
+}
+
+impl ActiveSqsSplits {
+    pub fn output_queue_names(&self) -> Vec<&str> {
+        self.queue_names
+            .values()
+            .map(|QueueNameUpdate { output_name, .. }| output_name.as_str())
+            .collect()
+    }
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Serialize, JsonSchema)]
+#[serde(rename_all = "camelCase")] // active_filters -> activeFilters
+pub struct WorkloadQueueRegistryStatus {
+    /// Optional even though it's currently the only field, because in the future there will be
+    /// fields for other queue types.
+    pub sqs_details: Option<ActiveSqsSplits>,
+}
+
+/// Defines a Custom Resource that holds a central configuration for splitting a queue. mirrord
+/// users specify a splitter by name in their configuration. mirrord then starts splitting according
+/// to the spec and the user's filter.
+#[derive(CustomResource, Clone, Debug, Deserialize, Serialize, JsonSchema)]
+#[kube(
+    group = "queues.mirrord.metalbear.co",
+    version = "v1alpha",
+    kind = "MirrordWorkloadQueueRegistry",
+    shortname = "qs",
+    status = "WorkloadQueueRegistryStatus",
+    namespaced
+)]
+pub struct MirrordWorkloadQueueRegistrySpec {
+    /// A map of the queues that should be split.
+    /// The key is used by users to associate filters to the right queues.
+    pub queues: BTreeMap<QueueId, SplitQueue>,
+
+    /// The resource (deployment or Argo rollout) that reads from the queues.
+    pub consumer: QueueConsumer,
+
+    /// These tags will be set for all temporary SQS queues created by mirrord for queues defined
+    /// in this MirrordWorkloadQueueRegistry, alongside with the original tags of the respective
+    /// original queue. In case of a collision, the temporary queue will get the value from the
+    /// tag passed in here.
+    pub tags: Option<HashMap<String, String>>,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Serialize, JsonSchema)]
+#[serde(rename = "SQSSessionStatus", rename_all = "camelCase")]
+pub struct SqsSessionStatus {
+    /// Queue ID -> old and new queue names.
+    pub queue_names: BTreeMap<QueueId, QueueNameUpdate>,
+
+    // A bit redundant, because the registry resource status has the mapping from env var name
+    // to queue id, and `queue_names` has the mapping from queue id to name update, but, saving
+    // it here in the form that is useful to reader, for simplicity and readability.
+    /// Env var name -> old and new queue names.
+    pub env_updates: BTreeMap<String, QueueNameUpdate>,
+    // TODO: add updates to config map references.
+}
+
+/// The [`kube::runtime::wait::Condition`] trait is auto-implemented for this function.
+/// To be used in [`kube::runtime::wait::await_condition`].
+pub fn is_session_ready(session: Option<&MirrordSqsSession>) -> bool {
+    session
+        .map(|session| session.status.is_some())
+        .unwrap_or_default()
+}
```

Comment:
I don't think this should be here, operator code seems like a better fit

Reply:
Really? But it depends on the definition of the status struct, and might have to be changed if we change that struct. I think having it here makes more sense and could also help people remember to change it if they change the struct.